processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.join.table;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.query.dimension.DefaultDimensionSpec;
+import org.apache.druid.query.dimension.DimensionSpec;
+import org.apache.druid.query.lookup.LookupExtractor;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.DimensionSelector;
+import org.apache.druid.segment.data.SingleIndexedInt;
+import org.apache.druid.segment.join.JoinConditionAnalysis;
+import org.apache.druid.segment.join.lookup.LookupJoinMatcher;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Map;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LookupJoinMatcherTest
+{
+  private final Map<String, String> lookupMap =
+      ImmutableMap.of("foo", "bar", "null", "", "empty String", "", "", "empty_string");
+  private static final String PREFIX = "j.";
+
+  @Mock
+  private LookupExtractor extractor;
+
+  @Mock
+  private ColumnSelectorFactory leftSelectorFactory;
+
+  @Mock
+  private DimensionSelector dimensionSelector;
+
+  private LookupJoinMatcher target;
+
+  @Before
+  public void setUp()
+  {
+    Mockito.doReturn(true).when(extractor).canIterate();
+    Mockito.doReturn(lookupMap.entrySet()).when(extractor).iterable();
+  }
+
+  @Test
+  public void testCreateConditionAlwaysFalse()
```

Comment:
Is the idea that this should not throw an exception? If so, perhaps add that to the test name or add a comment.

Reply:
Done